src/wagtail_live/publishers.py

```diff
@@ -0,0 +1,238 @@
+            In such case, the client side knows that new updates are available and sends a
+            GET request to get those updates.
+
+            (Http404) if a page corresponding to the channel_id given doesn't exist.
+        """
+
+        live_page = get_object_or_404(self.model, channel_id=channel_id)
+        response = JsonResponse(data={}, status=200)
+        response["Last-Update-At"] = live_page.last_update_timestamp
+        return response
+
+    def get(self, request, channel_id, *args, **kwargs):
+        """See base class."""
+
+        live_page = get_object_or_404(self.model, channel_id=channel_id)
+        last_update_client = self.get_last_update_client_from_request(request=request)
+        updated_posts, current_posts = live_page.get_updates_since(
+            last_update_ts=datetime.fromtimestamp(last_update_client),
+        )
+
+        return JsonResponse(
+            {
+                "updates": updated_posts,
+                "currentPosts": current_posts,
+                "lastUpdateTimestamp": live_page.last_update_timestamp,
+            }
+        )
+
+
+class LongPollingPublisher(PollingPublisherMixin):
+    """Long polling Publisher. Class Based View.
+
+    This class handles delivering new updates to the client side (Long polling technique).
+    It accepts 2 request methods (POST and GET) which correspond to the following steps:
+
+    1- The client side initiates (shake) the communication by sending a POST request.
+        The publisher acknowledges by sending relevant data to the client side.
+
+    2- The client side asks new updates by sending a GET request.
+        Server side keeps the connection open until a new update is available.
+        In that case, updates are directly sent to client side.
+        If the polling timeout duration is reached, it sends a response
+        containing a timeOutReached parameter which indicates the client side
+        that there aren't updates available.
+    """
+
+    url_name = "long-polling"
+
+    def post(self, request, channel_id, *args, **kwargs):
+        """See base class."""
+
+        live_page = get_object_or_404(self.model, channel_id=channel_id)
+        return JsonResponse(
+            {
+                "livePosts": [live_post.id for live_post in live_page.live_posts],
+                "lastUpdateTimestamp": live_page.last_update_timestamp,
+            }
+        )
+
+    def get(self, request, channel_id, *args, **kwargs):
+        """Sends updates when they are available as long as the polling timeout isn't reached.
+        See base class.
+        """
+
+        last_update_client = self.get_last_update_client_from_request(request=request)
+        starting_time = time.time()
+        polling_timeout = get_polling_timeout()
+
+        while time.time() - starting_time < polling_timeout:
+            live_page = get_object_or_404(self.model, channel_id=channel_id)
```

Comment:
At some point this might be optimized since when many people watch the live-stream they will essentially all be doing many identical queries to the database all the time. But database caching like django-cachalot will already help out in cases like this I think (only when you have considerably many users).

Reply:
Alright, do we support it out of the box and define a boolean setting like `WAGTAIL_LIVE_USE_CACHALOT` or shall we just mention in the docs that this optimisation is possible?

Reply:
I think we should just mention this possibility. If people use e.g. cachalot it will be automatically fixed, so no setting is required.

Reply:
Good.